### PR TITLE
adapting to new api

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Stable Names for datasets](https://trello.com/c/zJVzRWzc/852-stable-names-for-datasets)

File: geoengine/permissions.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def __repr__(self) -> str:
 class ResourceId:
     '''A wrapper for a resource id'''
 
-    def __init__(self, resource_type: Literal['DatasetId', 'Layer', 'LayerCollection'],
+    def __init__(self, resource_type: Literal['DatasetName', 'Layer', 'LayerCollection'],
```
Comment (michaelmattig) [inline geoengine/permissions.py:87]: I think it's still called `DatasetId` in the backend

File: tests/test_upload.py [hunk 2]
```diff
@@ -54,9 +54,9 @@ def test_upload(self):
             gdf = geopandas.GeoDataFrame(
                 df, geometry=geopandas.GeoSeries.from_wkt(polygons), crs="EPSG:4326")
 
-            dataset_id = ge.upload_dataframe(gdf)
+            dataset_name = ge.upload_dataframe(gdf)
 
-            self.assertEqual(dataset_id, DatasetId("fc5f9e0f-ac97-421f-a5be-d701915ceb6f"))
+            self.assertEqual(dataset_name, DatasetName("fc5f9e0f-ac97-421f-a5be-d701915ceb6f"))
```
Comment (michaelmattig) [inline tests/test_upload.py:59]: isn't this missing the namespace?

File: geoengine/workflow_builder/blueprints.py [hunk 3]
```diff
@@ -1,18 +1,11 @@
...
 def sentinel2_band(band_name, provider="5779494c-f3a2-48b3-8a2d-5fbba8c5b6c5", utm_zone="UTM32N"):
     '''Creates a workflow for a band from Sentinel 2 data.'''
-    band_source = operators.GdalSource(
-        dataset=api.ExternalDataId(
-            type="external",
-            providerId=provider,
-            layerId=f"{utm_zone}:{band_name}"
-        )
-    )
+    band_source = operators.GdalSource(f"_:{provider}:{utm_zone}:{band_name}")
```
Comment (michaelmattig) [inline geoengine/workflow_builder/blueprints.py:8]: I think this does not work. I get the following error when I try to register this workflow:

> 2023-06-06T12:16:08.698151Z  WARN Request{request_id=cfc0003f-7652-4a1d-90e0-62fcf78df514}: tracing_actix_web::middleware: Error encountered while processing the incoming HTTP request: ErrorResponse { error: "BodyDeserializeError", message: "named data must consist of at most three parts at line 1 column 127" }

Comment (ChristianBeilschmidt) [reply geoengine/workflow_builder/blueprints.py]: [444fd3f](https://github.com/geo-engine/geoengine-python/pull/136/commits/444fd3f24b6f2833b8d4df03cbb2d7c66add5da8)

File: tests/test_upload.py (hunk 2 above)
Comment (ChristianBeilschmidt) [reply tests/test_upload.py]: [9d605a1](https://github.com/geo-engine/geoengine-python/pull/136/commits/9d605a196010ee301d1bd4ca0bdc30b823bdf44a)

File: geoengine/permissions.py (hunk 1 above)
Comment (ChristianBeilschmidt) [reply geoengine/permissions.py]: I've changed it in the backend.

[81b1983](https://github.com/geo-engine/geoengine-python/pull/136/commits/81b1983e092036eb315c9628f2ca3f7e79e3fd66)

Comment (michaelmattig) [reply geoengine/permissions.py]: I tried to test it by running the add_public_raster_dataset.ipynb Notebook but I get an error on `get_xarray`:

> ServiceException: {"error":"UnableToParseQueryString","message":"Unable to parse query string: Invalid time 2014-04-01T12:00:00.000 00:00"}

Comment (ChristianBeilschmidt) [reply geoengine/permissions.py]: Remove your dependencies and reinstall the library. This has something to do with the library changes @jdroenner  did back then.

File: geoengine/workflow_builder/blueprints.py (hunk 3 above)
Comment (ChristianBeilschmidt) [reply geoengine/workflow_builder/blueprints.py]: For the providers, you can put stuff in backticks. Then, you can put arbitrary stuff in there, which will also help for the ebv stuff with JSON.
It does not change the limitation for dataset names, just for NamedData.

**Review by michaelmattig — APPROVED** [review]

**Pull request merged**